### PR TITLE
pull user roles authentication login form registration form

### PR DESCRIPTION
**Pull request by Killakan789**

### CODE REVIEW

File: symfony/config/packages/security.yaml [hunk 1]
```diff
@@ -30,10 +30,10 @@ security:
...
     access_control:
         # web version
-        - { path: '/customer', roles: [ IS_AUTHENTICATED_FULLY, ROLE_MANAGER ] }
-        - { path: '/customer/{id}', roles: [ IS_AUTHENTICATED_FULLY, ROLE_MANAGER ] }
-        - { path: '/customer/add', roles: [ IS_AUTHENTICATED_FULLY, ROLE_ADMIN ] }
-        - { path: '/customer/update/{id}', roles: [ IS_AUTHENTICATED_FULLY, ROLE_ADMIN ] }
+        - { path: '/customer', roles: [ IS_AUTHENTICATED_FULLY, ROLE_USER ] }
+        - { path: '/customer/{id}', roles: [ IS_AUTHENTICATED_FULLY, ROLE_USER ] }
+        - { path: '/customer/add', roles: [ IS_AUTHENTICATED_FULLY, ROLE_MANAGER ] }
+        - { path: '/customer/update/{id}', roles: [ IS_AUTHENTICATED_FULLY, ROLE_MANAGER ] }
```
Comment (Killakan789) [inline symfony/config/packages/security.yaml]: #7 #8 added and changed access rights in security yaml

File: symfony/src/Controller/CustomerController.php [hunk 2]
```diff
@@ -33,7 +33,7 @@ public function __construct(CustomerRepository $customerRepository,ValidatorInte
 
     /**
      * @Route("/customer/add", name="add_customer", methods={"POST"})
-     * @IsGranted("ROLE_ADMIN")
+     * @IsGranted("ROLE_MANAGER")
```
Comment (Killakan789) [inline symfony/src/Controller/CustomerController.php:37]: #7 #8 added and changed access rights in Controller

File: symfony/src/Controller/CustomerController.php [hunk 3]
```diff
@@ -22,6 +22,8 @@
 class CustomerController
 {
     private $customerRepository;
+    private $validator;
+    private $serializer;
```
Comment (Ez2815) [inline symfony/src/Controller/CustomerController.php:26]: also added defining injected services vars #6

File: symfony/config/routes.yaml [hunk 4]
```diff
@@ -26,14 +22,36 @@ delete:
...
 
-app_logout:
+logout:
   path: /logout
-  methods: GET
+  controller: App\Controller\SecurityController::logout
+  methods: GET|POST
+
+#api routes
+apiAdd:
```
Comment (avionwd) [inline symfony/config/routes.yaml]: Use the `snake_case` naming strategy for route names to follow the defined standards and agreements.

File: symfony/migrations/Version20210123180354.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        return '';
+    }
+
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('DROP TABLE product');
+        $this->addSql('DROP INDEX UNIQ_8D93D6497BA2F5EB ON user');
+        $this->addSql('ALTER TABLE user DROP api_token');
```
Comment (avionwd) [inline symfony/migrations/Version20210123180354.php:25]: Created and deleted column in one PR can be omitted as does not exist in the PROD database so only migrations are created and they useless.
The same situation for table `product`.

File: symfony/src/Entity/User.php [hunk 6]
```diff
@@ -42,6 +42,11 @@ class User implements UserInterface
      */
     private $username;
 
+     /**
+      * @ORM\Column(type="string", unique=true, nullable=true)
+      */
+     private $apiToken;
```
Comment (avionwd) [inline symfony/src/Entity/User.php]: This column was deleted from the database with the migration.

File: symfony/config/packages/security.yaml [hunk 7]
```diff
@@ -4,56 +4,39 @@ security:
         ROLE_ADMIN: [ ROLE_ADMIN ]
```
Comment (avionwd) [inline symfony/config/packages/security.yaml:3]: Role hierarchy should be used when complex or custom combinations of roles are used. So, no need to define all roles in the config.

File: symfony/config/packages/security.yaml [hunk 8]
```diff
@@ -4,56 +4,39 @@ security:
...
-            # https://symfony.com/doc/current/security/impersonating_user.html
-            # switch_user: true
-
-    # Easy way to control access for large sections of your site
-    # Note: Only the *first* access control that matches will be used
     access_control:
-        # or require ROLE_ADMIN or IS_AUTHENTICATED_FULLY for /admin*
-        - { path: '^/admin', roles: [ IS_AUTHENTICATED_FULLY, ROLE_ADMIN ] }
+        # web version
         - { path: '/customer', roles: [ IS_AUTHENTICATED_FULLY, ROLE_USER ] }
```
Comment (avionwd) [inline symfony/config/packages/security.yaml]: This is no correct use of the access_controll plugin. Read again the [documentation](https://symfony.com/doc/current/security.html#securing-url-patterns-access-control).

It says:

> The most basic way to secure part of your app is to secure an entire URL pattern in security.yaml. For example, to require ROLE_ADMIN for all URLs that start with /admin

So, you should only define the basic rule for API endpoint:
```
- { path: '^/customer', roles: [ IS_AUTHENTICATED_FULLY ] }
```

Role-based access checks already added in the controller action https://github.com/Killakan789/testcode/pull/13/files#diff-2b8ce039958e6c1929ce529f7496743d5dd7267505b862a69c0ab706bb9540e4R37

----
Helpful documentation:
- https://symfony.com/doc/current/security.html#securing-controllers-and-other-code
- https://symfony.com/doc/current/security/access_control.html

**Review by avionwd — CHANGES_REQUESTED** [review]
Remove obsolete classes that not used to solve the current task (additional authenticators).
How about the functional tests? Do you have a possibility to create tests to cover access settings?

File: symfony/config/routes.yaml (hunk 4 above)
Comment (Ez2815) [reply symfony/config/routes.yaml]: ok, will do

File: symfony/config/packages/security.yaml (hunk 7 above)
Comment (Ez2815) [reply symfony/config/packages/security.yaml:3]: okay will do.Actually i just tried to set up as much as possible (i wanted to just use the path with regex-"starts from" but thought that this should show that i'm trying to create as much as i can and understand that we may add different roles to access certain pages)will do that

File: symfony/config/routes.yaml [hunk 9]
```diff
@@ -10,7 +10,7 @@ get:
   path: /customer/{id}
   controller: App\Controller\CustomerController::get
 
-getAll:
+get_all:
   path: /customer/
   controller: App\Controller\CustomerController::getAll
```
Comment (Ez2815) [inline symfony/config/routes.yaml:15]: @avionwd  changed all paths with such name to snakecase names

File: symfony/config/packages/security.yaml [hunk 10]
```diff
@@ -30,11 +27,7 @@ security:
                 path: app_logout
     access_control:
         # web version
-        - { path: '/customer', roles: [ IS_AUTHENTICATED_FULLY, ROLE_USER ] }
-        - { path: '/customer/{id}', roles: [ IS_AUTHENTICATED_FULLY, ROLE_USER ] }
-        - { path: '/customer/add', roles: [ IS_AUTHENTICATED_FULLY, ROLE_MANAGER ] }
-        - { path: '/customer/update/{id}', roles: [ IS_AUTHENTICATED_FULLY, ROLE_MANAGER ] }
-        - { path: '/customer/delete/{id}', roles: [ IS_AUTHENTICATED_FULLY, ROLE_ADMIN ] }
+        - { path: '^/customer', roles: [ IS_AUTHENTICATED_FULLY ] }
```
Comment (Ez2815) [inline symfony/config/packages/security.yaml:30]: @avionwd  changed access control construction

**Review by Ez2815 — APPROVED** [review]

**Pull request merged**